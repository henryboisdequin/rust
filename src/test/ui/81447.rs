mod some_module {
    pub struct Test<T: ?Sized>(T);
}

use some_module::Test;

struct TestBuilder;

impl TestBuilder {
    fn build(self) -> Test {
        Test(self) 
        // ^~ERROR cannot initialize a tuple struct which contains private fields
    }
}

fn main() {}
