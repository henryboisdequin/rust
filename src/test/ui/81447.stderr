error[E0423]: cannot initialize a tuple struct which contains private fields
  --> $DIR/81447.rs:11:9
   |
LL |         Test(self) 
   |         ^^^^
   |
note: constructor is not visible here due to private fields
  --> $DIR/81447.rs:2:32
   |
LL |     pub struct Test<T: ?Sized>(T);
   |                                ^ private field

error[E0107]: missing generics for struct `Test`
  --> $DIR/81447.rs:10:23
   |
LL |     fn build(self) -> Test {
   |                       ^^^^ expected 1 type argument
   |
note: struct defined here, with 1 type parameter: `T`
  --> $DIR/81447.rs:2:16
   |
LL |     pub struct Test<T: ?Sized>(T);
   |                ^^^^ -
help: use angle brackets to add missing type argument
   |
LL |     fn build(self) -> Test<T> {
   |                           ^^^

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0107, E0423.
For more information about an error, try `rustc --explain E0107`.
